Makefile

```diff
@@ -30,11 +30,17 @@ all: test lint build
 
 # Run tests
 .PHONY: test
-export KUBEBUILDER_ASSETS := $(TOOLS_BIN_DIR)
+# Use envtest based on the version of kubernetes/client-go configured in the go.mod file.
+# If this version of envtest is not available yet, submit a PR similar to
+# https://github.com/kubernetes-sigs/kubebuilder/pull/2287 targeting the kubebuilder
+# "tools-releases" branch. Make sure to look up the appropriate etcd version in the
+# kubernetes release notes for the minor version you're building tools for.
+ENVTEST_VERSION = $(shell go list -m k8s.io/client-go | cut -d" " -f2 | sed 's/^v0\.\([[:digit:]]\+\)\.[[:digit:]]\+$$/1.\1.x/')
```

Comment:
`setup-envtest` downloads kb assets in a different location than what is expected (tmp/k8s/<os_specific_location>). We would have to set `KUBEBUILDER_ASSETS="$(shell $(ENVTEST) use $(ENVTEST_K8S_VERSION) -p path)` similar to [this](https://github.com/kubernetes-sigs/kubebuilder/pull/2272/files).

I tried testing the PR locally and it failed. 

```
      unable to start control plane itself: failed to start the controlplane. retried 5 times: fork/exec /usr/local/kubebuilder/bin/etcd: no such file or directory
  occurred
```
I am surprised on how it passed in the CI (maybe there is a bug in CI or a different envtest version is being considered).